rfcs/0124-worker-manager.md

```diff
@@ -233,6 +365,32 @@ strategy which can forward each request to an external service will be
 provided.  Providers do not need to be implemented directly inside of the
 worker manager, but can instead act as a bridge to an external service.
 
+### Example Bid Conversation
+This is a sample conversation, written as if each part of the system is a
+person speaking with each other.  This is an illustrative conversation and is
+not intended as a firm part of the design.
+
+* Worker Manager: Hey, Bidding strategy, you should run.
+* Bidding Strategy: Ok.  Queue, how many pending tasks?
+* Queue: 10.
+* Bidding Strategy (to all providers): How much running capacity do you have?
+* EC2 Provider: 2 running, 2 pending.
+* GCP Provider: 1 running.
+* Bidding Strategy (monologue): I see 3 running instances and 2 pending
+  instances.  The first five tasks already have workers, but I should create
+  capacity for the remaining five.
+* Bidding Strategy (to all providers): Can any of you provide me bids for creating
+  five capacity?
+* EC2 Provider: I can offer you 5 m3.medium instances for $2/h or 1 i5.8xlarge
+  for $20/h.  I think these are reliable, here's the details.
```

Comment:
What does "reliable" mean?  Reliably able to fulfill the bid, or the resulting instance is reliable?

Reply:
Reliable is sort of hard to define, but it's a gut feeling level evaluation.  In more concrete terms, this would be something based on outcomes of requests and instances, as well as possible status APIs which could give more info.  It's sort of hand-wavy and I don't want to specify in too much detail in the RFC because it's still not even defined for ec2-manager.

Reply:
It can still be handwavy, but should probably clarify that it's about the likelihood that the bid will succeed, and doesn't describe the resulting compute capacity.